runtime/queries/metricsview.go

```diff
@@ -304,3 +305,48 @@ func writeCSV(meta []*runtimev1.MetricsViewColumn, data []*structpb.Struct, writ
 
 	return nil
 }
+
+func writeXLSX(meta []*runtimev1.MetricsViewColumn, data []*structpb.Struct, writer io.Writer) error {
+	f := excelize.NewFile()
+	defer func() {
+		_ = f.Close()
+	}()
+
+	sw, err := f.NewStreamWriter("Sheet1")
+	if err != nil {
+		return err
+	}
+
+	headers := make([]interface{}, 0, len(meta))
+	for _, v := range meta {
+		headers = append(headers, v.Name)
+	}
+
+	if err := sw.SetRow("A1", headers, excelize.RowOpts{Height: 45, Hidden: false}); err != nil {
+		return err
+	}
+
+	row := make([]interface{}, 0, len(meta))
+	for i, s := range data {
+		for _, f := range s.Fields {
+			row = append(row, f.AsInterface())
```

Comment:
Just double-checking – it supports maps and lists being returned from `AsInterface()`?